tests/testShapeletPsfApproxPlugins.py

```diff
@@ -113,6 +119,97 @@ def testForced(self):
+        task = lsst.meas.base.SingleFrameMeasurementTask(config=config, schema=self.schema)
+        measCat = lsst.afw.table.SourceCatalog(self.schema)
+        measRecord = measCat.addNew()
+        psfImage = lsst.afw.image.ImageD(os.path.join(self.psfDir, "galsimPsf_0.9.fits"))
+        psfImage.setXY0(lsst.afw.geom.Point2I(0, 0))
+        kernel = lsst.afw.math.FixedKernel(psfImage)
+        psf = lsst.meas.algorithms.KernelPsf(kernel)
+        self.exposure.setPsf(psf)
+        center = lsst.afw.geom.Point2D(psfImage.getArray().shape[0]/2, psfImage.getArray().shape[1]/2)
+        measRecord.set(self.centroidKey, center)
+        task.run(measCat, self.exposure)
+        self.assertTrue(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_contains_nan"))
+        self.assertTrue(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_max_inner_iterations"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_max_outer_iterations"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_exception"))
+
+    def testOuterIterationsFlag(self):
+        config = lsst.meas.base.SingleFrameMeasurementTask.ConfigClass()
+        config.slots.centroid = None
+        config.slots.shape = None
+        config.slots.psfFlux = None
+        config.slots.apFlux = None
+        config.slots.instFlux = None
+        config.slots.modelFlux = None
+        config.slots.calibFlux = None
+        config.doReplaceWithNoise = False
+        config.plugins.names = ["modelfit_ShapeletPsfApprox"]
+        config.plugins["modelfit_ShapeletPsfApprox"].sequence = ["Full"]
+        config.plugins["modelfit_ShapeletPsfApprox"].models["Full"].optimizer.maxOuterIterations = 1
+        task = lsst.meas.base.SingleFrameMeasurementTask(config=config, schema=self.schema)
+        measCat = lsst.afw.table.SourceCatalog(self.schema)
+        measRecord = measCat.addNew()
+        psfImage = lsst.afw.image.ImageD(os.path.join(self.psfDir, "galsimPsf_0.9.fits"))
+        psfImage.setXY0(lsst.afw.geom.Point2I(0, 0))
+        kernel = lsst.afw.math.FixedKernel(psfImage)
+        psf = lsst.meas.algorithms.KernelPsf(kernel)
+        self.exposure.setPsf(psf)
+        center = lsst.afw.geom.Point2D(psfImage.getArray().shape[0]/2, psfImage.getArray().shape[1]/2)
+        measRecord.set(self.centroidKey, center)
+        task.run(measCat, self.exposure)
+        self.assertTrue(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_contains_nan"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_max_inner_iterations"))
+        self.assertTrue(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_max_outer_iterations"))
+        self.assertFalse(measRecord.get("modelfit_ShapeletPsfApprox_Full_flag_exception"))
```

Comment:
Maybe refactor these three test functions a bit, so they can share some code?  I could imagine having a new function that does all or most of the common setup stuff, so you can just make each test function contain what's unique to that test.